src/main/java/org/commcare/suite/model/QueryPromptValidation.java

```diff
@@ -0,0 +1,40 @@
+package org.commcare.suite.model;
+
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.javarosa.core.util.externalizable.ExtWrapTagged;
+import org.javarosa.core.util.externalizable.Externalizable;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.javarosa.xpath.expr.XPathExpression;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+public class QueryPromptValidation implements Externalizable {
+
+    private XPathExpression xpath;
+    private String message;
```

Comment:
should this not be a `Text` attribute to support translation?

Also commented on the spec: https://docs.google.com/document/d/1wF9SFfX7-R86cPRgabrqbTU9BaGdp3IlAYlTjInWXTU/edit?disco=epp37nri35q